js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,372 @@
+              method: 'DELETE',
+            })
+            .then(() => window.location.reload());
+
+          app.modal.show(LoadingModal);
+        }
+      };
+
+      items.add(
+        'uninstall',
+        <Button icon="fas fa-trash-alt" className="Button Button--primary" onclick={uninstall.bind(this)}>
+          {app.translator.trans('core.admin.extension.uninstall_button')}
+        </Button>
+      );
+    }
+
+    return items;
+  }
+
+  infoItems() {
+    const items = new ItemList();
+
+    if (this.extension.authors) {
+      let authors = [];
+
+      Object.keys(this.extension.authors).map((author) => {
+        authors.push(this.extension.authors[author].name);
+      });
+
+      items.add('authors', [icon('fas fa-user'), <span>{authors.join(', ')}</span>]);
+    }
+
+    if (this.extension.source || this.extension.support) {
+      items.add(
+        'source',
+        LinkButton.component(
+          {
+            href: this.extension.source ? this.extension.source.url : this.extension.support.source,
+            icon: 'fas fa-code',
+            external: true,
+            target: '_blank',
+          },
+          app.translator.trans('core.admin.extension.info_links.source')
+        )
+      );
+    }
+
+    if (this.extension.extra['flarum-extension'].info) {
+      Object.keys(this.infoFields).map((field) => {
+        if (this.extension.extra['flarum-extension'].info[field]) {
+          items.add(
+            field,
+            LinkButton.component(
+              {
+                href: this.extension.extra['flarum-extension'].info[field],
+                icon: this.infoFields[field],
+                external: true,
+                target: '_blank',
+              },
+              app.translator.trans(`core.admin.extension.info_links.${field}`)
+            )
+          );
+        }
+      });
+    }
+
+    return items;
+  }
+
+  submitButton() {
+    return (
+      <Button onclick={this.saveSettings.bind(this)} className="Button Button--primary" loading={this.loading} disabled={!this.changed()}>
+        {app.translator.trans('core.admin.settings.submit_button')}
+      </Button>
+    );
+  }
+
+  /**
+   * getSettings accepts an object of settings where the key is the settings key.
+   * Depending on the type of input, you can set the type to 'bool', 'select', or
+   * any standard <input> type.
+   *
+   * @example
+   * {
+   *   'acme.checkbox': {
+   *      label: app.translator.trans('acme.admin.setting_label')
+   *      type: 'bool'
+   *   }
+   * }
+   *
+   * @example
+   * {
+   *   'acme.select': {
+   *      label: app.translator.trans('acme.admin.setting_label'),
+   *      type: 'select',
+   *      options: {
+   *        'option1': 'Option 1 label',
+   *        'option2': 'Option 2 label',
+   *      },
+   *      default: 'option1',
+   *   }
+   * }
+   *
+   * @param settings
+   * @returns {JSX.Element}
+   */
+  getSettings(settings) {
```

Comment:
I still have mixed feelings about storing settings as an object. See my comment on `ExtensionData` for more info.